### DISCUSSION TITLE
grapql-ws protocol

### DISCUSSION THREAD

**Discussion by kristjanvalur** [strawberry-graphql/strawberry#1623] [Q&A] [answered]
Hi there.
I'm using strawberry as a reference to write a custom client application to do subscriptions in graphql.

I'm confused about the actual websocket protocols.

1. First, the documentation here https://strawberry.rocks/docs/general/subscriptions#graphql-over-websocket-protocols seems contraditory, it seems to say that `graphql-ws` is the old protocol.  I think someone got the names and the links wrong in this section.
2. I've arrived at https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md for the actual protocol specification, but I see discrepcancies.  The linked protocol uses a message `type: "subscribe"` whereas both **strawberry** and **GraphiQL** Seem to use  `type: "start"` instead.  
3. I'm starting with code originally written for AWS, https://docs.aws.amazon.com/appsync/latest/devguide/real-time-websocket-client.html, and it _says_ its using grapql-ws, but it uses a diferent payload again, using a `data` field instead of a `query` field.

Any thoughs on this?  What is the actual graphql spec to use to work with standard servers these days?

(I tried passing a "subscribe" type message to strawberry and it just ignores it, doesn't even respond, whereas it should probably close the websocket with error 4400, according to the spec.)

Cheers, and thanks for any help.

**Comment by DoctorJohn** [comment, marked as answer]
Take a look at the protocol specs you linked. Under "Communication" it says:

> The WebSocket sub-protocol for this specification is: graphql-transport-ws

In other words: `graphql-ws` is the [legacy protocol](https://github.com/apollographql/subscriptions-transport-ws/blob/master/PROTOCOL.md) and `graphql-transport-ws` is the [new protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md). It's confusing since the new protocols repository is named graphql-ws but the actual protocol is called graphql-transport-ws.

**Reply by kristjanvalur** [reply]
Thank you.  Reading the doc with a comb again, I finally noticed the quote.
Additionally, `GraphiQL` is currently using the `graphql-ws` and I was assuming it was using the latest spec.
